### PR TITLE
[SPARK-48777][BUILD] Properly lint, vet and check for license headers.

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
This PR fixes a number of build issues to make sure that the source code is properly formatted and adheres to the Apache rules for license header requirements.

### Why are the changes needed?
Compliance

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing Tests

### CODE REVIEW

File: cmd/spark-connect-example-spark-session/main.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"flag"
 	"log"
 
+	"github.com/apache/spark-connect-go/v35/spark/sql/utils"
```
Comment (mathiasschw-db) [inline cmd/spark-connect-example-spark-session/main.go:24]: There should not be spaces between these imports

File: spark/client/channel/channel_test.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ func TestBasicChannelParsing(t *testing.T) {
 	assert.ErrorIs(t, err, sparkerrors.InvalidInputError)
 
 	cb, err = channel.NewBuilder(goodChannelURL)
+	assert.Nilf(t, err, "Should not have an error for a proper URL")
```
Comment (mathiasschw-db) [inline spark/client/channel/channel_test.go:61]: `assert.NoError` is the right one to use here

File: spark/sql/utils/check.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import "fmt"
+
+func WarnOnError(f func() error) {
+	if err := f(); err != nil {
+		fmt.Println("Received error:", err)
```
Comment (mathiasschw-db) [inline spark/sql/utils/check.go:23]: You should not use fmt.Println from within library code. At the very minimum, use a logger but do consider a slightly more general version where you can pass in an error handler (which can then in turn be something that logs

File: cmd/spark-connect-example-spark-session/main.go (hunk 1 above)
Comment (grundprinzip) [reply cmd/spark-connect-example-spark-session/main.go:24]: This comes from the gofumpt rules serparating the std library from external packages

image

Comment (mathiasschw-db) [reply cmd/spark-connect-example-spark-session/main.go:24]: Apologies for being a bit imprecise. I mean this import and the two below that also not std lib and are from the same library

Comment (grundprinzip) [reply cmd/spark-connect-example-spark-session/main.go:24]: Will fix that.

File: spark/client/channel/channel_test.go (hunk 2 above)
Comment (grundprinzip) [reply spark/client/channel/channel_test.go:61]: Done

File: spark/sql/utils/check.go (hunk 3 above)
Comment (grundprinzip) [reply spark/sql/utils/check.go]: Done, added a callback.

**Review by zhengruifeng — APPROVED** [review]

**Pull request merged**